Ginger/Ginger/Actions/ActionEditPages/Communication/ActeMailEditPage.xaml.cs

```diff
@@ -74,7 +81,59 @@ private void Bind()
             }
             ShowDisplayNameOption();
         }
+        private void xcbValidationRequired_checked(object sender, RoutedEventArgs e)
+        {
+            IsValidationRequired.IsChecked = true;
+            CertificateStackPanel.Visibility = Visibility.Visible;
+        }
+        private void xcbValidationRequired_unchecked(object sender, RoutedEventArgs e)
+        {
+            IsValidationRequired.IsChecked = false;
+            CertificateStackPanel.Visibility = Visibility.Collapsed;
+        }
+        private void BrowseButton_Certificate(object sender, RoutedEventArgs e)
+        {
+            System.Windows.Forms.OpenFileDialog dlg = new System.Windows.Forms.OpenFileDialog();
+
+            dlg.DefaultExt = "*.crt";
+            dlg.Filter = "CRT Files (*.crt)|*.crt";
+            string SolutionFolder = WorkSpace.Instance.Solution.Folder.ToUpper();
+            if (dlg.ShowDialog() == System.Windows.Forms.DialogResult.OK)
+            {
+
+                string FileName = dlg.FileName.ToUpper();
+                if (FileName.Contains(SolutionFolder))
+                {
+                    FileName = FileName.Replace(SolutionFolder, @"~\");
+                }
+                FileName = WorkSpace.Instance.SolutionRepository.ConvertFullPathToBeRelative(FileName);
+                xcbCertificatePathTextBox.Text = FileName;
+                string targetPath = System.IO.Path.Combine(SolutionFolder, @"Documents\EmailCertificates\Certificates");
+                if (!System.IO.Directory.Exists(targetPath))
+                {
+                    System.IO.Directory.CreateDirectory(targetPath);
+                }
 
+                string destFile = System.IO.Path.Combine(targetPath, System.IO.Path.GetFileName(FileName));
+
+                int fileNum = 1;
+                string copySufix = "_Copy";
+                while (System.IO.File.Exists(destFile))
+                {
+                    fileNum++;
+                    string newFileName = System.IO.Path.GetFileNameWithoutExtension(destFile);
+                    if (newFileName.IndexOf(copySufix) != -1)
+                        newFileName = newFileName.Substring(0, newFileName.IndexOf(copySufix));
+                    newFileName = newFileName + copySufix + fileNum.ToString() + System.IO.Path.GetExtension(destFile);
+                    destFile = System.IO.Path.Combine(targetPath, newFileName);
+                }
+
+                System.IO.File.Copy(FileName, destFile, true);
+                xcbCertificatePathTextBox.Text = @"~\Documents\EmailCertificates\Certificates\" + System.IO.Path.GetFileName(destFile);
```

Comment:
create a seperate variable for path and refer that in all places where path is being used